DOL.WHD.Section14c.Web/src/modules/components/accountManagementControls/accountFormController.test.js

```diff
@@ -2,17 +2,194 @@ describe('accountFormController', function() {
 
     beforeEach(module('14c'));
 
-    beforeEach(inject(function ($rootScope, $controller) {
+    beforeEach(inject(function ($rootScope, $controller, $location, _$q_, apiService) {
         scope = $rootScope.$new();
+        $q = _$q_;
+        mockApiService = apiService;
+        mockRouteParams = {};
+        mockLocation = $location;
+
+        getAccount = $q.defer();
+        spyOn(mockApiService, 'getAccount').and.returnValue(getAccount.promise);
+    
+        getRoles = $q.defer();
+        spyOn(mockApiService, 'getRoles').and.returnValue(getRoles.promise);
+
+        modifyAccount = $q.defer();
+        spyOn(mockApiService, 'modifyAccount').and.returnValue(modifyAccount.promise);
+
+        createAccount = $q.defer();
+        spyOn(mockApiService, 'createAccount').and.returnValue(createAccount.promise);
 
         accountFormController = function() {
             return $controller('accountFormController', {
-                '$scope': scope
+                '$scope': scope,
+                '$routeParams': mockRouteParams,
+                '$location.': mockLocation
             });
         };
     }));
 
     it('invoke controller', function() {
         var controller = accountFormController();
     });
+
+    it('when editing user forVals get set', function() {
+        var controller = accountFormController();
+        getAccount.resolve({ data: { } });
+        scope.$apply();
+
+        expect(scope.formVals).not.toBe(undefined);
+    });    
+
+    it('when editing user if the getAccount service has an error show loading error', function() {
+        var controller = accountFormController();
+        getAccount.reject({ data: { error: 'error'} });
+        scope.$apply();
+
+        expect(controller.loadingError).toBe(true);
+    });    
+
+    it('when editing user if the getAccount service has an error show loading error, no error logging', function() {
+        var controller = accountFormController();
+        getAccount.reject({ data: { error: ''} });
+        scope.$apply();
+
+        expect(controller.loadingError).toBe(true);
+    });          
+
+    it('default formVals if creating a new account', function() {
+        mockRouteParams = { userId: 'create'};
+        var controller = accountFormController();
+        scope.$apply();
+
+        expect(scope.formVals).not.toBe(undefined);
```

Comment:
I left the assertion since this is just testing that an empty object is created.   I also added one to insure the roles array is there.   https://github.com/AppliedIS/dol-whd-14c/commit/eec34a6d4188192f954e3c81d7e4f3f4ba4b111b